File: java/org/gessinger/gepard/xmp/Requester.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;

import java.util.List ;

public class Requester
{
  static public void main ( String[] args )
  {
    try
    {
      Requester j = new Requester() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  Requester()
  throws Exception
  {
    final Client client = Client.getInstance() ;

    String name = "getFileList" ;
    System.out.println ( "Request data for name=" + name ) ;

    client.request ( name, new ResultCallback()
    {
      public void result ( Event e )
      {
        if ( e.isBad() )
        {
          System.out.println ( "e.getStatusReason()=" + e.getStatusReason() ) ;
        }
        else
        {
          List<String> list = (List<String>) e.getBodyValue ( "file_list" ) ;
          System.out.println ( Util.toString ( list ) ) ;
        }
        client.close() ;
      }
    }) ;
  }
}

File: python/xmp/Requester.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )

from gepard import Event, Client

client = Client.getInstance()
# client.setDaemon ( True )

def on_close ( err, info ):
	print ( err )
def on_error ( err, info ):
	print ( err )
def on_shutdown ( err, info ):
	print ( "shutdown called" )

client.onClose ( on_close )
client.onError ( on_error )
client.onShutdown ( on_shutdown )

def getFileList ( e ):
	print ( "callback for get getFileList" )
	if e.isBad():
		print ( e.getStatusReason() )
	else:
		print ( e.getValue ( "file_list" ) )
	e.getClient().close()

client.request ( "getFileList", getFileList )

